operators/nvidia_video_codec/nv_video_decoder/nv_video_decoder.hpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef NV_VIDEO_DECODER_NV_VIDEO_DECODER_HPP
+#define NV_VIDEO_DECODER_NV_VIDEO_DECODER_HPP
+
+#include <memory>
+#include <string>
+
+#include <cuda.h>
+#include "holoscan/core/gxf/entity.hpp"
+#include "holoscan/core/operator.hpp"
+#include "holoscan/utils/cuda_stream_handler.hpp"
+
+#include "FFmpegDemuxer.h"
+#include "NvDecoder/NvDecoder.h"
+
+namespace holoscan::ops {
+class StreamDataProvider : public FFmpegDemuxer::DataProvider {
+ public:
+  StreamDataProvider() {}
+  ~StreamDataProvider() {}
+
+  void SetData(uint8_t* data, int size) {
+    offset_ = 0;
+    data_ = data;
+    size_ = size;
+  }
+  // Fill in the buffer owned by the demuxer/decoder
```

Comment:
Is this update correct, or could you please clarify the relation to the demux/decoder buffer here?
```suggestion
  // Fill in the buffer owned by the demuxer/decoder and advance the offset
```